From 2937d5d460677fa6c92c09798d4c71c441bd9070 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Wed, 28 Dec 2022 18:50:43 +0100
Subject: [PATCH 2/2] Manual revert "gstomx: remove
 gst_omx_buffer_set_omx_buf/get_omx_buf"

Needed to fix build after Manual revert of "omxbufferpool: refactor to allow memory sharing"
---
 omx/gstomx.c | 18 ++++++++++++++++++
 omx/gstomx.h |  3 +++
 2 files changed, 21 insertions(+)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 65dd577..647c890 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -70,6 +70,8 @@ static GHashTable *core_handles;
 G_LOCK_DEFINE_STATIC (buffer_flags_str);
 static GHashTable *buffer_flags_str;
 
+static GQuark gst_omx_buffer_data_quark = 0;
+
 GstOMXCore *
 gst_omx_core_acquire (const gchar * filename)
 {
@@ -3873,6 +3875,20 @@ gst_omx_set_default_role (GstOMXClassData * class_data,
     class_data->component_role = default_role;
 }
 
+void
+gst_omx_buffer_set_omx_buf (GstBuffer * buffer, GstOMXBuffer * omx_buf)
+{
+  gst_mini_object_set_qdata (GST_MINI_OBJECT_CAST (buffer),
+      gst_omx_buffer_data_quark, omx_buf, NULL);
+}
+
+GstOMXBuffer *
+gst_omx_buffer_get_omx_buf (GstBuffer * buffer)
+{
+  return gst_mini_object_get_qdata (GST_MINI_OBJECT_CAST (buffer),
+      gst_omx_buffer_data_quark);
+}
+
 static void
 _class_init (gpointer g_class, gpointer data)
 {
@@ -4043,6 +4059,8 @@ plugin_init (GstPlugin * plugin)
   GST_DEBUG_CATEGORY_INIT (OMX_API_TRACE, "OMX_API_TRACE", 0,
       "gst-omx performace");
 
+  gst_omx_buffer_data_quark = g_quark_from_static_string ("GstOMXBufferData");
+
   /* Read configuration file gstomx.conf from the preferred
    * configuration directories */
   env_config_dir = g_strdup (g_getenv (*env_config_name));
diff --git a/omx/gstomx.h b/omx/gstomx.h
index 79c2e23..77b5099 100644
--- a/omx/gstomx.h
+++ b/omx/gstomx.h
@@ -486,6 +486,9 @@ gboolean          gst_omx_buffer_import_fd (GstOMXBuffer * buffer, GstBuffer * i
 
 void              gst_omx_set_default_role (GstOMXClassData *class_data, const gchar *default_role);
 
+void              gst_omx_buffer_set_omx_buf (GstBuffer * buffer, GstOMXBuffer * omx_buf);
+GstOMXBuffer *    gst_omx_buffer_get_omx_buf (GstBuffer * buffer);
+
 /* refered by plugin_init */
 GST_DEBUG_CATEGORY_EXTERN (gst_omx_video_debug_category);
 
-- 
2.34.1

